lightkube/codecs.py

```diff
@@ -71,6 +124,8 @@ def load_all_yaml(stream: Union[str, TextIO], context: dict = None, template_env
         will be used during templating.
     * **template_env** - `jinja2` template environment to be used for templating. When absent a standard
         environment is used.
+    * **client** - (Optional) A lightkube.Client used to infer the Resource definition for any 
```

Comment:
This parameter is not very obvious and can be confusing, I'd rename it to `infer_from_client`, or something that tells me what is going on. It can be something more elegant if you have a better idea :)

Reply:
I'm thinking about this, how's the client important to infer the resource definition?